src/components/DetailsView/DetailsView.js

```diff
@@ -1,59 +1,42 @@
-import React, { Component, createRef } from 'react';
+import { useRef, useState } from 'react';
 import ReactDOM from 'react-dom';
+import { usePokemons } from '../../context/PokemonsProvider';
 import BackButton from '../BackButton/BackButton';
 import { Card } from '../Card';
 import { Details } from '../Details';
 import { Overlay } from '../Overlay';
 import './DetailsView.css';
 
-class DetailsView extends Component {
-	constructor( props ) {
-		super( props );
-		this.detailsViewRef = createRef( null );
+export default function DetailsView() {
+	const detailsViewRef = useRef( null );
+	const { setCurrentPokemonId, currentPokemon } = usePokemons();
+	const [ hasOverlay, setHasOverlay ] = useState( true );
 
-		this.state = {
-			hideOverlay: false,
-			pokemon: props.pokemon,
-		};
-
-		this.handleBackClick = this.handleBackClick.bind( this );
-		this.setPokemonData = this.setPokemonData.bind( this );
+	if ( ! currentPokemon ) {
+		return null;
 	}
 
-	// handle back button clicking
-	handleBackClick() {
-		this.setState( {
-			hideOverlay: true,
-		} );
-
-		this.detailsViewRef.current.classList.add( 'hidden' );
+	// Handle modal close.
+	const closeModal = () => {
```

Comment:
useCallback (prevent the <Overlay> component re-render).

Reply:
Done